### PR TITLE
Create docs/platforms, relocate android & web pages

### LINKED ISSUE #472: Create docs/platforms and move Android and web pages there
We now have an **_Kotlin_ Android Quick start** page (#471), but it doesn't make sense to put it under 

- **Languages** > **Android Java**

At least one other Android page is on its way (via **Flutter**). It seems like it's time to create `docs/platforms`.

### PR DESCRIPTION
**Pull request by chalin**
Closes #472 -- see that issue for some context, also note the comment below this one.

Preview:

- https://deploy-preview-473--admiring-torvalds-f1ffea.netlify.app/docs/
- https://deploy-preview-473--admiring-torvalds-f1ffea.netlify.app/docs/languages/
- https://deploy-preview-473--admiring-torvalds-f1ffea.netlify.app/docs/platforms/

### CODE REVIEW

**Comment by chalin** [comment]
Note that this PR make minimal changes, and _only_ relocates the Android and Web pages without changing them. After this PR is merged, the plan would be to move the current Android pages (which are Java specific) to a sublevel, like this:

- /docs/
  - platforms/
    - android/
      - **java/{quick-start, basics,...}**
      - kotlin/quick-start
      - dart or flutter ...
    - web/{quick-start, basics, ...}

Above, I also show where the Kotlin and Dart/Flutter pages would go.

File: content/docs/_index.md [hunk 1]
```diff
@@ -13,11 +13,22 @@ material for all [supported languages](languages/).
...
 
-  [Select a language](languages/) and then choose its **Quick start**.
+  Select a [language][] or [platform][], then choose its **Quick start**.
 
 - Interested in **gRPC feature details?**
 
   Try one of the following:
 
-  - [Select a language](languages/) and then choose **Tutorial** or **API reference**
+  - Select a [language][] or [platform][], then choose **Tutorial** or **API reference**
```
Comment (mjpitz) [inline content/docs/_index.md:22]: This should just need to be `[lanugage] or [platform]`.

**Review by mjpitz — APPROVED** [review]
Looks good to me. I think the links could be simplified form `[label][]` to `[label]` but not the end of the world as the empty brackets don't seem to appear on the rendered pages.

File: content/docs/_index.md (hunk 1 above)
Comment (chalin) [reply content/docs/_index.md:22]: If you write `[text][]` then some editors (like VS Code) give edit-time validation that your markdown link is defined (by underlining it, in the case of VS Code). Very useful feature. As you point out, it doesn't affect the link generation.

Comment (mjpitz) [reply content/docs/_index.md:22]: Neat! I'm going to have to start doing this

**Comment by chalin** [comment]
@ejona86 @thisisnotapril - any feedback on this? Or did you want to discuss it during our next meeting?

File: content/docs/_index.md (hunk 1 above)
Comment (ejona86) [reply content/docs/_index.md:22]: `[text][]` is part of original markdown. I didn't realize `[text]` worked. That sounds like a markdown extension, which is why some editors might not notice it.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**